src/commands/init.ts

```diff
@@ -0,0 +1,116 @@
+import { Command, flags } from "@oclif/command";
+import { execSync } from "child_process";
+import * as fs from "fs";
+
+const DEFAULT_PROJECT_NAME = "react";
+
+/**
+ * @description a script that is used in the `package.json` scripts
+ * @example "g:component": "eb-scripts generate -t react-component -n"
+ */
+type Scripts = {
+  /* A script */
+  [script in string]: string;
+};
+
+/**
+ * @description valid project type
+ * @example `react`
+ * @todo eventually, we'll add others like `react-native`, `react-typescript`, `node`, etc.
+ */
+type ProjectType = "react";
+
+/**
+ * @description the available options for generate scripts that may be added to a user's `package.json`
+ */
+type ScriptByProject = {
+  [project in ProjectType]: Scripts;
+};
+
+// Grab the path of the user's project
+const pathWhereScriptIsRunning = process.cwd();
+const scriptsByProject: ScriptByProject = {
+  react: {
+    "g:component": "eb-scripts generate -t react-component -n"
+  }
+};
+export default class Init extends Command {
+  static description =
+    "initializes project by installing `eb-scripts` and adding scripts to `package.json`";
+
+  static examples = [`$ npx eb-scripts init -p react`];
+
+  static flags = {
+    help: flags.help({ char: "h" }),
+    // flag with a value (-p, --project=VALUE)
+    project: flags.string({
+      char: "p",
+      description: "language/framework of project"
+    })
+  };
+
+  static args = [
+    {
+      name: "project",
+      required: true,
+      description: "The language or framework of the project",
+      default: "react",
+      options: ["react"] as ProjectType[]
+    }
+  ];
+
+  async run() {
+    const { flags } = this.parse(Init);
+    const project = flags.project || DEFAULT_PROJECT_NAME;
+    // Grab their package Json
+    const packageJsonLocation = `${pathWhereScriptIsRunning}/package.json`;
+    const packageJson = require(packageJsonLocation);
+    const theirScripts = packageJson.scripts;
+    const validProjectTypes = Object.keys(scriptsByProject);
+    const isValidProject = validProjectTypes.includes(project);
+
+    // Check if project is valid
+    if (!isValidProject) {
+      console.error(`
+        Error: invalid project type.
+        Please use one of the following:
+        ${validProjectTypes}
+        `);
+      return;
+    }
+
+    this.log(`Initializing new ${project} project`);
+
+    // Otherwise, it is valid, so we tell TypeScript to assert it as a ProjectType
+    const projectScripts: Scripts = scriptsByProject[project as ProjectType];
+
+    // Combine their scripts with our scripts
+    const updatedScripts = { ...theirScripts, ...projectScripts };
```

Comment:
Should this spread be the other way around? ie take our defaults and then apply the user scripts?

Reply:
Keeping it as is, results in this:

![image](https://user-images.githubusercontent.com/3806031/67970255-ecb72500-fbc7-11e9-86a6-fe4cd9d6461b.png)

Switching the spread around as you suggest, results in this:

![image](https://user-images.githubusercontent.com/3806031/67970358-1708e280-fbc8-11e9-8d2a-016c539ee0e8.png)

I think I originally wrote it as is so that our script would be added at the bottom. And if we do it this way, we can overwrite their `g:component` script - which is what we want I think

![image](https://user-images.githubusercontent.com/3806031/67970517-59caba80-fbc8-11e9-9774-019dbda42185.png)

But I could be convinced otherwise!